modules/xfem/src/userobjects/XFEMRankTwoTensorMarkerUserObject.C

```diff
@@ -0,0 +1,114 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "XFEMRankTwoTensorMarkerUserObject.h"
+
+#include "libmesh/quadrature.h"
+#include "RankTwoTensor.h"
+#include "RankTwoScalarTools.h"
+#include "Assembly.h"
+#include <limits>
+
+template <>
+InputParameters
+validParams<XFEMRankTwoTensorMarkerUserObject>()
+{
+  InputParameters params = validParams<XFEMMarkerUserObject>();
+  params.addClassDescription(
+      "Mark elements to be cut by XFEM based on a scalar extracted from a RankTwoTensor");
+  params.addParam<MooseEnum>(
+      "scalar_type",
+      RankTwoScalarTools::scalarOptions(),
+      "Scalar quantity to be computed from tensor and used as a failure criterion");
+  params.addRequiredParam<std::string>("tensor", "The material tensor name.");
+  params.addRequiredCoupledVar("threshold", "The threshold for crack growth.");
+  params.addRequiredParam<bool>(
+      "average", "Should the tensor quantity be averaged over the quadrature points?");
+  params.addParam<Point>(
+      "point1",
+      Point(0, 0, 0),
+      "Start point for axis used to calculate some cylindrical material tensor quantities");
+  params.addParam<Point>(
+      "point2",
+      Point(0, 1, 0),
+      "End point for axis used to calculate some cylindrical material tensor quantities");
+  return params;
+}
+
+XFEMRankTwoTensorMarkerUserObject::XFEMRankTwoTensorMarkerUserObject(
+    const InputParameters & parameters)
+  : XFEMMarkerUserObject(parameters),
+    _tensor(getMaterialProperty<RankTwoTensor>(getParam<std::string>("tensor"))),
+    _scalar_type(getParam<MooseEnum>("scalar_type")),
+    _point1(parameters.get<Point>("point1")),
+    _point2(parameters.get<Point>("point2")),
+    _threshold(coupledValue("threshold")),
+    _average(getParam<bool>("average")),
+    _JxW(_assembly.JxW()),
+    _coord(_assembly.coordTransformation())
+{
+}
+
+bool
+XFEMRankTwoTensorMarkerUserObject::doesElementCrack(RealVectorValue & direction)
+{
+  bool does_it_crack = false;
+  unsigned int numqp = _qrule->n_points();
+
+  if (_average)
+  {
+    Real average_threshold;
```

Comment:
You aren't initializing this Real! The first assignment is a `+=` which could give you garbage.